modules/ppcp-api-client/src/Entity/Order.php

```diff
@@ -237,7 +229,7 @@ static function ( PurchaseUnit $unit ): array {
 			$order['application_context'] = $this->application_context()->to_array();
 		}
 		if ( $this->payment_source() ) {
-			$order['payment_source'] = $this->payment_source()->to_array();
+			$order['payment_source'] = $this->payment_source();
```

Comment:
:thinking: I think the result of `to_array` is not supposed to contain objects like this?

Reply:
I agree and don't understand the reason for this change. I'll leave for @Dinamiko to review.

Reply:
`PaymentSource` was refactored it seems, the new implementations does not have `to_array()` but I think it should.

Reply:
It could be that `payment_source` is not needed here. So it either needs to be converted into an array or removed from this output.

It seems like `Order::to_array` is only used in two places where we output it in our ajax response or JS data. So we don't send it to the API and don't notice that this value is weird. Should be ok to remove this line then.

Reply:
Yeah, I know, in this kind of abandoned branch I did it like this https://github.com/woocommerce/woocommerce-paypal-payments/tree/PCP-1801-experience-context/modules/ppcp-api-client/src/Entity/PaymentSource

But it needs to be either converted into an array or removed from `to_array` since a class instance automatically converted into JSON would not be useful anyway (I think it will be just `{}` by default).